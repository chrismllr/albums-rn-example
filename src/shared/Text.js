// @flow
import glamorous from 'glamorous-native';
import { palette, fontSize, spacing } from '../styles/common';

type BaseTextProps = {
  themeColor: string,
  fontSize: number,
  fontWeight: string,
  marginBottom: number
};

export const BaseText = glamorous.text({
  color: palette.gray.dark,
  fontSize: fontSize.fs4,
  marginBottom: spacing.sp2
}, ({ themeColor, fontSize: size, fontWeight, marginBottom }: BaseTextProps) => ({
  color: palette[themeColor],
  fontSize: size,
  fontWeight,
  marginBottom
}));
