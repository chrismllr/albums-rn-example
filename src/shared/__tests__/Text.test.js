import React from 'react';
import 'react-native';
import { BaseText } from '../Text';
import renderer from 'react-test-renderer';

describe('BaseText', () => {
  it('renders to snapshot', () => {
    const tree = renderer.create(
      <BaseText />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
