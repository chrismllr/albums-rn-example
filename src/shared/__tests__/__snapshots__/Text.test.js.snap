// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseText renders to snapshot 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#333",
        "fontSize": 16,
        "marginBottom": 6,
        "textAlign": "center",
      },
      Object {
        "color": undefined,
        "fontSize": undefined,
      },
    ]
  }
/>
`;
